app/main/controllers/upload/RTMediaUploadView.php

```diff
@@ -38,7 +38,11 @@ public function __construct( $attr ) {
 	public static function upload_nonce_generator( $echo = true, $only_nonce = false ) {
 
 		if ( $echo ) {
-			wp_nonce_field( 'rtmedia_upload_nonce', 'rtmedia_upload_nonce' );
+			/**
+			 * If we use wp_nonce_field, it'll create a hidden field with ID, which will generate HTML warning.
+			 * Because this nonce is being used in multiple places on the same page.
+			 */
+			echo '<input type="hidden" name="rtmedia_upload_nonce" value="' . esc_attr( wp_create_nonce( 'rtmedia_upload_nonce' ) ) . '" />';
```

Comment:
Do we need this now?

Reply:
The ajax upload of activity comment is handled in BuddyPress. So we'll have to keep individual nonce for each upload form.